source/backend/dal/Repositories/PropertyRepository.cs

```diff
@@ -464,6 +465,74 @@ public HashSet<long> GetMatchingIds(PropertyFilterCriteria filter)
             return query.Select(p => p.PropertyId).ToHashSet();
         }
 
+        /// <summary>
+        /// Return a summary List of Management activities for a especific property.
+        /// </summary>
+        /// <param name="propertyId"></param>
+        /// <returns>List of Property's management activities.</returns>
+        public IList<PimsPropPropActivity> GetManagementActivitiesByProperty(long propertyId)
+        {
+            User.ThrowIfNotAllAuthorized(Permissions.ManagementView);
+
+            List<PimsPropPropActivity> activities = Context.PimsPropPropActivities.AsNoTracking()
+                    .Include(x => x.PimsPropertyActivity)
+                        .ThenInclude(y => y.PropMgmtActivityTypeCodeNavigation)
+                    .Include(x => x.PimsPropertyActivity)
+                        .ThenInclude(y => y.PropMgmtActivitySubtypeCodeNavigation)
+                    .Include(x => x.PimsPropertyActivity)
+                       .ThenInclude(y => y.PropMgmtActivityStatusTypeCodeNavigation)
+                    .Where(p => p.PropertyId == propertyId)
+                    .ToList();
+
+            return activities;
+        }
+
+        /// <summary>
+        /// Get the Property Management Activity by Id.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public PimsPropPropActivity GetManagementActivityById(long id)
+        {
+            User.ThrowIfNotAllAuthorized(Permissions.ManagementView);
+
+            return Context.PimsPropPropActivities
+                        .AsNoTracking()
+                        .Include(x => x.PimsPropertyActivity)
```

Comment:
Do you need the includes on the code navigation properties as well?

Reply:
I needed it for the Service to check that the activity status is 'not started'.